jmxtrans-core/src/main/java/com/googlecode/jmxtrans/cli/JmxTransConfiguration.java

```diff
@@ -144,10 +169,138 @@ public File apply(String input) {
 	@Getter @Setter
 	private int resultProcessorExecutorWorkQueueCapacity = 100000;
 
+	private static final String USE_SEPARATE_EXECUTORS_PROPERTY = "use.separate.executors";
 	@Parameter(
 			names = {"--use-separate-executors"},
 			description = "If this set every server node will be handed by separate executor."
 	)
 	@Getter @Setter
 	private boolean useSeparateExecutors = false;
+
+	private static abstract class PropertySetter<T> {
+		private final String key;
+		private final Class<T> type;
+		public PropertySetter(String key, Class<T> type) {
+			this.key = key;
+			this.type = type;
+		}
+		public void setValue(TypedProperties typedProperties, JmxTransConfiguration configuration) {
+			T value = typedProperties.getTypedProperty(key, type);
+			if (value != null) {
+				doSetValue(value, configuration);
+			}
+		}
+
+		protected abstract void doSetValue(T value, JmxTransConfiguration configuration);
+	}
+
+	private static PropertySetter[] SETTERS = new PropertySetter[]{
+			new PropertySetter<Boolean>(CONTINUE_ON_ERROR_PROPERTY, Boolean.class) {
```

Comment:
This would be soo much cleaner with lambda. Is anyone still using Java 7 (I'm loosing touch with enterprise realities)

Reply:
I saw a problem with sortpom plugin and Java 7 on my machine, I wonder whether it stills compile on Java 7